test/functional/test_framework/p2p.py

```diff
@@ -539,8 +539,16 @@ def send_and_ping(self, message, timeout=60):
         self.send_message(message)
         self.sync_with_ping(timeout=timeout)
 
-    # Sync up with the node
+    def sync_send_with_ping(self, timeout=60):
+        """Ensure any to-be-received message on this connection is processed by SendMessages"""
+        # Calling sync_with_ping twice requires that the node calls
+        # `ProcessMessage` twice, and thus ensures `SendMessages` must have
+        # been called at least once
+        self.sync_with_ping()
+        self.sync_with_ping()
+
     def sync_with_ping(self, timeout=60):
+        """Ensure any sent message on this connection is processed by ProcessMessages"""
```

Comment:
thanks for adding description here & in `sync_send_with_ping`, but imo it could be clearer. offering a suggestion incase you agree this captures the mechanism better:

"Ensure the recipient node has run ProcessMessages for all previously sent p2p messages"

Reply:
Isn't ''**any** message" too strong (instead of "at least one message"), especially here but also above?  `ProcessMessages`, `ProcessMessage` and `SendMessages`  can all return early. With `SendMessages`, it goes together with a disconnect, but for `ProcessMessage(s)` there are many other ways, so that in case of multiple messages not all might have been processed in this iteration of the loop.

Reply:
you make a good point, I was making an assumption about test setup that isn't necessarily true.

Reply:
Tried to fix, thanks